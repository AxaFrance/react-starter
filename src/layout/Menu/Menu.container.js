import React from 'react';
import { useLocation } from 'react-router-dom';
import { Menu } from './Menu';

const EnhancedMenu = () => {
  const location = useLocation();
  return <Menu location={location} />;
};

export default EnhancedMenu;
