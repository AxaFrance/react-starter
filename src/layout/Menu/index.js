export { Menu } from './Menu';
export { default } from './Menu.container';
