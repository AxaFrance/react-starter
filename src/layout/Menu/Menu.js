import React from 'react';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import { Link } from 'react-router-dom';
import { NavBar, NavBarItem } from '@axa-fr/react-toolkit-all';

import './Menu.scss';

export const Menu = ({ location }) => (
  <NavBar positionInit={-1} onClick={() => {}} isVisible>
    <NavBarItem
      classModifier={classNames({
        active: location && location.pathname === '/',
      })}
      actionElt={
        <Link className="af-nav__link" to="/">
          Accueil
        </Link>
      }
    />
    <NavBarItem
      classModifier={classNames({
        active: location && location.pathname === '/dashboard',
      })}
      actionElt={
        <Link className="af-nav__link" to="/dashboard">
          Dashboard
        </Link>
      }
    />
  </NavBar>
);

const propTypes = {
  location: PropTypes.object.isRequired,
};

Menu.propTypes = propTypes;
