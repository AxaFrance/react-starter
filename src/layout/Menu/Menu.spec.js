import React from 'react';
import { render } from '@testing-library/react';
import { StaticRouter } from 'react-router-dom';
import { Menu } from './Menu';

const location = {
  pathname: '/whatever',
};

const wrapper = ({ children }) => (
  <StaticRouter context={{}}>{children}</StaticRouter>
);

it('1. Renders Menu component without crashing', () => {
  const { asFragment } = render(<Menu location={location} />, { wrapper });
  expect(asFragment()).toMatchSnapshot();
});
