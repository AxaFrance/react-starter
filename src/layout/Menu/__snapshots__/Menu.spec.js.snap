// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`1. Renders Menu component without crashing 1`] = `
<DocumentFragment>
  <div
    class="af-nav-container"
  >
    <button
      aria-controls="mainmenu"
      aria-haspopup="true"
      class="mask fade show"
      id="togglemenu"
      type="button"
    />
    <nav
      aria-label="Menu principal"
      class="af-nav af-drawer left show"
      role="navigation"
    >
      <div
        class="container af-nav__wrapper"
      >
        <div
          class="af-nav__aside-header"
        >
          <h4
            class="af-nav__aside-title"
          >
            Menu
          </h4>
          <button
            aria-controls="mainmenu"
            aria-haspopup="true"
            class="af-nav__aside-close"
            id="togglemenu"
            type="button"
          >
            Ã—
          </button>
        </div>
        <ul
          aria-labelledby="togglemenu"
          class="af-nav__list "
          id="mainmenu"
          role="menubar"
        >
          <li
            class="af-nav__item "
            role="none"
          >
            <a
              aria-expanded="false"
              class="af-nav__link"
              href="/"
              role="menuitem"
              tabindex="-1"
            >
              Accueil
            </a>
          </li>
          <li
            class="af-nav__item "
            role="none"
          >
            <a
              aria-expanded="false"
              class="af-nav__link"
              href="/dashboard"
              role="menuitem"
              tabindex="-1"
            >
              Dashboard
            </a>
          </li>
        </ul>
      </div>
    </nav>
  </div>
</DocumentFragment>
`;
